ql/src/java/org/apache/hadoop/hive/ql/stats/BasicStatsTask.java

```diff
@@ -118,11 +119,17 @@ public String getName() {
     private boolean isMissingAcidState = false;
     private BasicStatsWork work;
     private boolean followedColStats1;
+    private Map<String, String> providedBasicStats;
 
     public BasicStatsProcessor(Partish partish, BasicStatsWork work, HiveConf conf, boolean followedColStats2) {
       this.partish = partish;
       this.work = work;
       followedColStats1 = followedColStats2;
+      Table table = partish.getTable();
+      if (table.isNonNative()) {
```

Comment:
what about non-iceberg non-native tables ?
I think we can count the rows in a non-native table with the basic TS rowcounter

I think you are trying to identify your usecase here by `isNonNative` - I think you would be better off saving your initial decision about how you plan to execute this task in the `work`